refactor: handle errors

liammulh opened the pull request:
This PR:
- adds error handling via Vue's [error handler](https://vuejs.org/api/application.html#app-config-errorhandler) (see `main.ts`
- wraps the HTTP requests in `fillCache` in a try-catch
- adds an `alertMessage` utility along with docs and tests

Now the frontscope should `window.alert` an error message if an error occurs during:
- Component renders
- Event handlers
- Lifecycle hooks
- setup() function (Vue, not p5)
- Watchers
- Custom directive hooks
- Transition hooks
- HTTP requests in `fillCache` (currently the only place where we make HTTP requests)

liammulh commented:
For https://github.com/numberscope/frontscope/issues/63.

gwhitney commented:
All looks good, but I would suggest that a change of this sort should be accompanied by something in the docs, maybe a brief mention toward the end of `doc/user_guide.md` (am OK if there's someplace else you think it would make more sense).

liammulh commented:
In the most recent commits, I:

- added a note about `alertMessage` to `user_guide.md` and a note to `making-a-visualizer.md`,
- removed file extensions again since they crept back in (I think my IDE adds them when it auto-imports things),
- changed the name of the test for `alertMessage` from `alertError.spec.ts` to `alertMessage.spec.ts`,
- fixed some errors I saw in `CONTRIBUTING.md` (typo, broken links), and
- added a link to the `making-a-visualizer.md` doc in the `CONTRIBUTING.md` doc.

I believe https://github.com/numberscope/frontscope/pull/185#issuecomment-1307561503 has been addressed, @gwhitney.

liammulh commented:
By the way, I apologize for the shoddy Git commit history. I suppose this is one of the advantages of squash and merge. :)

gwhitney commented:
Adding new points/steps in CONTRIBUTING.md is outside the scope of this PR. I am not going to make you take it out and file a separate PR, but as a professional habit, you need to avoid blurring the focus of PRs. So please don't do that in future PRs. Also I slightly amended the clause about emailing (esp. as I am not sure where that email goes...)
Merging
